service/src/main/kotlin/xyz/lebkuchenfm/domain/xsounds/XSoundsFileRepository.kt

```diff
@@ -1,5 +1,11 @@
 package xyz.lebkuchenfm.domain.xsounds
 
+import com.github.michaelbull.result.Result
+
 interface XSoundsFileRepository {
-    suspend fun uploadXSoundFile(soundName: String, byteArray: ByteArray): String
+    suspend fun uploadXSoundFile(soundName: String, byteArray: ByteArray): Result<String, FileStorageError>
+}
+
+sealed interface FileStorageError {
```

Comment:
I feel like `FileStorageError` is too generic in this case. This should be about uploading new sound file.
```suggestion
sealed interface UploadXSoundFileError {
```